### PR TITLE
Improve developer workflow and add basic repository structure

### PR DESCRIPTION
**Pull request by jwilander**

### CODE REVIEW

File: source/developer/developer-setup.md [hunk 1]
```diff
@@ -217,23 +217,28 @@ Optionally, you can make that account a System Admin with the following command:
 go run mattermost.go -assign_role -email="user@example.com" -role="system_admin system_user"
 ```
 
-## Development Flow 
+## Development Flow
 
-A few tips to develop efficiently: 
+Here's a general workflow a Mattermost developer working on our [platform repository](https://github.com/mattermost/platform) will follow:
```
Comment (jasonblais) [inline source/developer/developer-setup.md]: `will follow` --> `follows` for active tense

Comment (jwilander) [inline source/developer/developer-setup.md]: Good call, changing..

**Comment by jasonblais** [comment]
Overall great work, love the "Repository Structure" section! 

Wondering if we should break the Development Flow into its own .html page? I feel it's kind of hidden below the Developer Machine instructions. 

For instance, if I was setting up my dev machine for Mac, not sure I'd scroll all the way down (unless I needed to see the Troubleshooting section).

We should also reference the Development Flow in our [code contribution guidelines](https://docs.mattermost.com/developer/contribution-guide.html), maybe under the `Install Mattermost and set up a Fork` section.

**Comment by jwilander** [comment]
Agreed, how about I move it to it's own page and I can add a link to it form the contribution guidelines? It looks like it is already linked to properly from here https://docs.mattermost.com/guides/developer.html and the sidebar

**Comment by jasonblais** [comment]
Yeah, that sounds good to me :+1:

**Comment by jwilander** [comment]
Ok @jasonblais, changes made. Please re-review at your leisure

**Pull request merged**